src/locate/helpers/helper.py

```diff
@@ -0,0 +1,186 @@
+        silhouette_avg = silhouette_score(X, cluster_labels)
+        silhouette_avgs[n] = silhouette_avg
+    key_max = max(silhouette_avgs.keys(), key=(lambda k: silhouette_avgs[k]))
+    return key_max
+
+def scaling(data):
+    '''
+    Normalizes data
+    '''
+    scaled_data = StandardScaler().fit_transform(data)
+    return scaled_data
+
+def kmeans_algorithm(data, sensor_number=None):
+    '''
+    Clustering data using K-Means Model
+    '''
+    if sensor_number == None:
+        sensor_number = silhouette(data)
+    
+    X = data[['properties.lat', 'properties.long', 'properties.population_density', 'properties.household_density', 
+              'properties.charcoal_per_km2', 'properties.firewood_per_km2', 'properties.grass_per_km2', 
+              'properties.wasteburning_per_km2', 'properties.kitch_outsidebuilt_per_km2',
+              'properties.kitch_makeshift_per_km2', 'properties.kitch_openspace_per_km2']]
+    
+    X_scaled = scaling(X)    
+    
+    kmeans = KMeans(n_clusters=sensor_number).fit(X_scaled) 
+    y_kmeans = kmeans.fit_predict(X_scaled)
+    
+    data_copy = data.copy()
+    
+    data_copy['cluster'] = y_kmeans
+    
+    kmeans_samples = data_copy.sample(frac=1).reset_index(drop=True)
+    kmeans_samples= kmeans_samples.drop_duplicates('cluster', keep = 'last')
+    kmeans_samples = kmeans_samples[['properties.district', 'properties.subcounty', 'properties.parish',  'properties.lat', 
+    'properties.long', 'geometry.coordinates']]
+    return json.loads(kmeans_samples.to_json(orient = 'records'))
+
+def get_data(data):
+ """
+ This function deserializes an JSON object.
+:param data: JSON data
+ :type data: str
+ """
+ json_data = json.loads(data)
+ print("Deserialized data: {}".format(data))
+ return json_data
+
+def point_exists_in_polygon(point, polygon):
+    '''
+    checks whether a given point exists in a defined polygon
+    '''
+    
+    geo_point = Point(point[0], point[1])
+    geo_polygon = {
+        'coordinates': polygon, 
+        'type': "Polygon"
+    }
+    geo_polygon = shape(geo_polygon)
+    if geo_polygon.contains(geo_point):
+        return True
+    else:
+        return False
+
+def delete_keys_from_dict(dictionary, keys):
+    '''
+    deletes certain keys and their values from a dictionary
+    '''
+    keys_set = set(keys) 
+    modified_dict = {}
+    for key, value in dictionary.items():
+        if key not in keys_set:
+            if isinstance(value, MutableMapping):
+                modified_dict[key] = delete_keys_from_dict(value, keys_set)
+            else:
+                modified_dict[key] = value  
+    return modified_dict
+
+def recommend_locations(sensor_number, must_have_coordinates, polygon):
+    '''
+    recommends parishes in which to place sensors
+    '''
+    parish = Parish()
+    if polygon==None:
+        return jsonify({'response': 'Please draw a polygon'}), 200
+    elif must_have_coordinates==None:
```

Comment:
1. There are some code duplications in here (`src/locate/helpers/helper.py`). 
- Lines 129- 142
- Lines 149-165
- Lines 166-186
- Lines 85-120

2. Additionally, we need to delete this file (`src/ml-service/packages/sensor_location/locate/modules/locate_helper.py`) from our codebase since it no longer serves a purpose.

Reply:
I don't see the duplications. Please elaborate

Reply:
lets continue this discussion from here: https://github.com/airqo-platform/AirQo-api/issues/108